### ISSUE TITLE
New UI/Form Standards?

### ISSUE THREAD

**Issue by excitedbox** [w3c/htmlwg#30]
How many more ui kits will we build until we develop a standard for the wider ecosystem of ui components?
Panels, cards, pickers of all kinds, trees, etc. The mess of ui kits we deal with only causes duplication and increases integration workload while limiting compatibility with functionality frameworks. (React, Angular etc.)

I think we have had long enough to see what UI functionality is most needed since 99% of Ui kits contain the same ~50 components. Isn't it time that we standardize these? It seems the whole reason for the web component standard to exist is to allow devs to add the missing ui components from the current HTML standards.

**Comment by coderextreme** [comment]
I agree.  I’m presently building out a custom 3D multi-select menu (actually, just a toggle button template) because I wasn’t aware of alternatives for X3D.  I suggest we start with a 3D/XR ui kit standard first, then do SVG, HTML, native and terminal.

https://www.web3d.org/x3d/content/examples/HumanoidAnimation/FacialAnimation/FacialAnimationComparisonScriptsIndex.html

AFAIK, web components kind of failed.

The thing is, we had standards like OpenLook and Motif (MoOLit), and a nascent CORBA-based system called Fresco, and the web set us way back.  All the managers loved the web, and the developers were like f* it.   If you want to start over with Fresco, it’s likely on to the freedesktop.org X11 distribution somewhere.  I was able to get it running recently.  The Fresco developers moved to Java, and we know where that went.

So don’t mind me while I build out a 3D ui kit built on open standards.

**Comment by coderextreme** [comment]
Beyond ui kits, I think we can standardize the file browser, product selection, product configuration, contacts/dating/meeting, 3D asset management, IT modeling, and more into a higher level “app” kit.   Respond if you want to work on this with me.  I have a few prototypes.  https://github.com/coderextreme/productconfig https://github.com/coderextreme/abstractionmapper

I’m guessing Lotus Notes, shopify and salesforce are in this space.
